Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -525,8 +525,10 @@ void HTMLInputElement::updateType()
     bool didRespectHeightAndWidth = m_inputType->shouldRespectHeightAndWidthAttributes();
     bool wasSuccessfulSubmitButtonCandidate = m_inputType->canBeSuccessfulSubmitButton();
 
-    if (didStoreValue && !willStoreValue && hasDirtyValue())
-        setAttributeWithoutSynchronization(valueAttr, AtomString { std::exchange(m_valueIfDirty, { }) });
+    if (didStoreValue && !willStoreValue && hasDirtyValue()) {
+        if (auto dirtyValue = std::exchange(m_valueIfDirty, { }); !dirtyValue.isEmpty())
+            setAttributeWithoutSynchronization(valueAttr, AtomString { WTFMove(dirtyValue) });
+    }
```

Comment:
Since we are checking `!isEmpty()`, we don’t need the `hasDirtyValue()` check; can just take that out. Also wondering if any other call sites making `hasDirtyValue()` checks also want to handle empty dirty values differently.